README.md

```diff
@@ -50,29 +50,454 @@ resolvers += "jitpack" at "https://jitpack.io"
+fancyHeadLine("I like tomatoes.")
+// <h1 style="border-bottom: 1px dashed tomato;">I like tomatoes.</h1>
+```
+
+
+#### Transforming Components
+Components are immutable, we can only modify them by creating a changed copy. Like you may know from Scalatags, you can call `.apply(...)` on any `VNode`, *append* more modifiers and get a new `VNode` with the applied changes back.
+
+```scala
+val a = div("dog")
+a(title := "the dog")
+// <div title="the dog">dog</div>
+```
+
+This can be useful for reusing html snippets.
+
+```scala
+val box = div(width := "100px", height := "100px")
+
+div(
+  box(backgroundColor := "powderblue"),
+  box(backgroundColor := "mediumseagreen"),
+)
+
+// <div>
+//  <div style="width: 100px; height: 100px; background-color: powderblue;"> </div>
+//  <div style="width: 100px; height: 100px; background-color: mediumseagreen;"></div>
+// </div>
+```
+
+Since modifiers are *appended*, they can overwrite existing ones. This is useful to adjust existing components to your needs.
+
+```scala
+val box = div(width := "100px", height := "100px")
+box(backgroundColor := "mediumseagreen", width := "200px")
+// <div style="width: 200px; height: 100px; background-color: mediumseagreen;"></div>
+```
+
+You can also *prepend* modifiers. This can be useful to provide defaults retroactively.
+
+```scala
+def withBorderIfNotProvided(vnode: VNode) = vnode.prepend(border := "3px solid coral")
+div(
+  withBorderIfNotProvided(div("hello", border := "7px solid moccasin")),
+  withBorderIfNotProvided(div("hello")),
+)
+// <div>
+//   <div style="border: 7px solid moccasin;">hello</div>
+//   <div style="border: 3px solid coral;">hello</div>
+// </div>
+```
+
+Source Code: [VDomModifier.scala](src/main/scala/outwatch/dom/VDomModifier.scala#L146)
+
+
+#### Use-Case: Flexbox
+When working with [Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/), you can set styles for the **container** and **children**. With `VNode.apply()` you can have all flexbox-related styles in one place. The child-components don't have to know anything about flexbox, even though they get specific styles assigned.
+
+```scala
+val itemA = div("A", backgroundColor := "mediumseagreen")
+val itemB = div("B", backgroundColor := "tomato")
+
+div(
+  height := "100px",
+  border := "1px solid black",
+
+  display.flex,
+
+  itemA(flexBasis := "50px"),
+  itemB(alignSelf.center),
+)
+// <div style="height: 100px; border: 1px solid black; display: flex;">
+//   <div style="background-color: mediumseagreen; flex-basis: 50px;">A</div>
+//   <div style="background-color: tomato; align-self: center;">B</div>
+// </div>
+```
+
+
+### Dynamic Content
+To visualize updates, use an `Observable[VDomModifier]` as if it was a `VDomModifier`.
+
+```scala
+import monix.reactive.Observable
+import concurrent.duration._
+    
+val counter = Observable.interval(1 second)
+div("count: ", counter)
+```
+
```

Comment:
Would be great to have live examples of this inside some box :)

Reply:
You are right. There should at least be a link to a jsfiddle.

Reply:
Maybe we could think of using something like `sbt-microsite` it's pretty easy to set up :)

Reply:
Yes, even better with a combination of compiling examples like https://github.com/scalameta/mdoc. You can add that to the documentation issue, to address this in a future change.